src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,351 @@
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled.button`
+  &&& {
+    background-color: ${styles.backgroundColor};
+    border-radius: ${(props) => props.theme.spacings.xxsmall};
+    border: ${styles.border};
+    width: ${(props) => (props.block ? '100%' : '')};
+  }
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const color = `${variantColor || 'primary'}`;
+
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw new Error(
+      `Error in Button: expected \`children\` of type \`string\` but found ${typeof children}`,
+    );
+  }
+
+  return (
+    <Flex
+      flex={block ? 1 : 0}
+      flexDirection="row"
+      alignItems="center"
+      alignSelf={styles.align({ align })}
+      justifyContent="center"
+    >
+      <Size height={styles.height({ size, theme })}>
+        <Space padding={styles.padding({ size, children, theme })}>
+          <StyledButton
+            color={color}
+            onClick={onClick}
+            variantColor={variantColor}
+            disabled={disabled}
+            size={size}
+            block={block}
+            variant={variant}
+            {...automation('ds-button')}
+          >
+            <React.Fragment>
+              {icon && iconAlign === 'left' && (
+                <Icon
```

Comment:
The icon seems to have a different color than the text in stories.
<img width="1196" alt="Screenshot 2020-04-13 at 12 29 00 PM" src="https://user-images.githubusercontent.com/9150371/79099712-63b13600-7d82-11ea-8912-611e8497469c.png">

Reply:
Its because Icon component is not working properly

Reply:
I've raised a PR for that.

Reply:
https://github.com/razorpay/blade/pull/97

Reply:
merged